src/shared/install.c

```diff
@@ -3242,6 +3301,12 @@ static int read_presets(RuntimeScope scope, const char *root_dir, UnitFilePreset
         assert(scope < _RUNTIME_SCOPE_MAX);
         assert(presets);
 
+        if (getpid_cached() == 1) {
+                r = proc_cmdline_parse(parse_proc_cmdline_item, &ps, 0);
```

Comment:
here too, i don't like that we parse the kernel cmdline *again* here. This should be done where we parse the kernel cmdline, and then be stored in global fields, which we just pass in here as extra arguments.

I think install.c should not be the place we do special handling in depending on what our PID is. In particular as this shouldn't be subject to PID=1 anyway, but rather to being invoked from a context where we made sure of RUNTIME_SCOPE_RUNNING (so that --test invocations take this into account too)